docs_src/design/adr/0005-Service-Self-Config.md

```diff
@@ -0,0 +1,131 @@
+- --confdir or -c (the directory where the configuration file is found - ./res is used by default if the confdir is not specified, where "." is the convention on Linux/Unix/MacOS which means current directory) 
+- --registry or -r (string indicating use of the registry)
+
+The distinction of command line options versus configuration will be important later in this ADR.
+
+Two command line options (-o for overwrite and -r for registry) are not overridable by environmental variables.
+
+NOTES:
+- For backwards compatibility, there is an edgex_registry environment variable that will override the  --configProvider / -cp and --registry / -r command line options.
+- Use of the --overwrite command line option should be used sparingly and with expert knowledge of EdgeX; in particular knowledge of how it operates and where/how it gets its configuration on restarts, etc.  Ordinarily, --overwrite is provided as a means to support development needs.  Use of --overwrite permanently in production enviroments is highly discouraged.
+
+### Configuration Initialization
+Each service has (or shall have if not providing it already) a local configuration file.  The service may use the local configuration file on initialization of the service (aka bootstrap of the service) depending on command line options and environmental variables (see below) provided at startup.
+
+**Using a configuration provider**
+
+When the configuration provider _is_ specified, the service will call on the configuration provider (Consul) and check if the top-level (root) namespace for the service exists.  If configuratation at the top-level (root) namespace exists, it indicates that the service has already populated its configuration into the configuration provider in a prior startup.
+
+If the service finds the top-level (root) namespace is already populated with configuration information it will then read that configuration information from the configuration provider under namespace for that service (and ignore what is in the local configuration file).
+
+If the service finds the top-level (root) namespace is not populated with configuration information, it will read its local configuration file and populate the configuration provider (under the namespace for the service) with configuration read from the local configuration file.
+
+A configuration provider can be specified with a command line argument (the -cp / --configProvider) or environment variable (the EDGEX_CONFIGURATION_PROVIDER environmental variable which overrides the command line argument).
+> NOTE:  the environmental variables are typically uppercase but there have been inconsistencies in environmental variable casing and changing it would involve non-backward compatible change.  This should be considered and made consistent in a future major release.
```

Comment:
This not can be removed. @lenny-intel do you agree?

Reply:
Naw, I think it is still relevant. We have edgex_registry, EDGEX_CONFIGURATION_PROVIDER, etc.

Reply:
The first part of the sentence is correct, however this statement isn't...

_and changing it would involve non-backward compatible change._ 

But we did change it, I thought it was originally specified as lowercase, and we changed it to uppercase because it hadn't yet been used in a production release!

Reply:
this a nit though and doesn't dramatically change what the ADR says... If you want to leave it so be it.